utils/__tests__/getDataUtils.spec.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  parseCategory,
+  getCategories,
+  getAllPhrases,
+  getKidsCategory,
+  getPhraseById,
+  fetchRawDictionary,
+  fetchDictionary,
+  CategoryDataObject,
+  DictionaryDataObject,
+  PhraseDataObject,
+  Phrase,
+} from '../getDataUtils';
+import { CountryVariant } from '../locales';
+import fetch, { Response } from 'node-fetch';
+
+jest.mock('node-fetch');
+
+// Create mock data for testing
+
+const mockCountryVariantCZ = 'cz' as CountryVariant;
+
+const mockPhraseData: PhraseDataObject = {
+  id: 'testPhrase',
+  main: {
+    sound_url: 'https://example.com/sound.mp3',
+    translation: 'translation',
+    transcription: 'transcription',
+  },
+  source: {
+    sound_url: 'https://example.com/sound_source.mp3',
+    translation: 'translation_source',
+    transcription: 'transcription_source',
+  },
+  image_url: 'https://example.com/image.jpg',
+};
+
+const mockCategoryData: CategoryDataObject = {
+  id: 'testCategory',
+  name: {
+    source: 'categoryNameSource',
+    main: 'categoryNameMain',
+  },
+  description: 'description',
+  phrases: ['testPhrase'],
+  hidden: true,
+};
+
+const mockDictionaryData: DictionaryDataObject = {
+  source: 'uk',
+  main: mockCountryVariantCZ,
+  categories: [mockCategoryData],
+  phrases: { testPhrase: mockPhraseData },
+};
+
+describe('Get Data Utils', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('parseCategory', () => {
+    it('should parse a CategoryDataObject into a Category', () => {
+      const parsedCategory = parseCategory(mockCategoryData, mockDictionaryData);
+
+      expect(parsedCategory.id).toEqual(mockCategoryData.id);
+      expect(parsedCategory.nameMain).toEqual(mockCategoryData.name.main);
+      expect(parsedCategory.nameUk).toEqual(mockCategoryData.name.source);
+      expect(parsedCategory.translations[0].getId()).toEqual(mockPhraseData.id);
+    });
+  });
+
+  describe('getCategories', () => {
+    it('should return an array of Category objects', () => {
+      const categories = getCategories(mockDictionaryData);
+      expect(categories.length).toEqual(1);
```

Comment:
The `1` seems needlessly hardcoded, why not `mockDictionaryData.categories.length` ?